src/plugins/intel_gpu/src/kernel_selector/cl_kernels/activation_ref.cl

```diff
@@ -32,7 +36,7 @@ KERNEL(activation)(
     const unsigned x = get_global_id(0);
     const uint y = (uint)get_global_id(1) % OUTPUT_SIZE_Y;
     const uint z = (uint)get_global_id(1) / OUTPUT_SIZE_Y;
-    #if OUTPUT_BATCH_NUM == 1
+    #if OUTPUT_BATCH_NUM_CONST == 1
```

Comment:
You can't use runtime value in preprocessing stage
Also, it will make preprocessor always choose only `else` branch (even in static scenario), since `OUTPUT_BATCH_NUM_CONST` doesn't exist from preprocessor's point of you
So, the proper solution will be keep it as it was. And in that case static path may be optimized, but in dynamic case we need to calculate batch index anyway (`else` branch)